hal/halx86/generic/x86bios.c

```diff
@@ -413,20 +450,35 @@ x86BiosCall(
     _Inout_ PX86_BIOS_REGISTERS Registers)
 {
     const ULONG StackBase = 0x2000;
-    FAST486_STATE EmulatorContext;
+    BOOLEAN Success = FALSE;
     ULONG FlatIp;
     PUCHAR InstructionPointer;
 
-    /* Initialize the emulator context */
-    Fast486Initialize(&EmulatorContext,
-                      x86MemRead,
-                      x86MemWrite,
-                      x86IoRead,
-                      x86IoWrite,
-                      x86BOP,
-                      x86IntAck,
-                      NULL,  // FpuCallback,
-                      NULL); // Tlb
+    if (InterlockedIncrement((PLONG)&HalpBiosCallCount) > 1)
```

Comment:
```suggestion
    if (InterlockedIncrement((PLONG)&HalpBiosCallCount) != 1)
```

Reply:
I'll use InterlockedIncrementUL.

Reply:
```
2023-08-03T12:44:58.5975579Z ../src/hal/halx86/generic/x86bios.c: In function 'x86BiosCall':
2023-08-03T12:44:58.6018413Z ../src/hal/halx86/generic/x86bios.c:466:13: error: implicit declaration of function 'InterlockedIncrementUL'; did you mean 'InterlockedIncrement'? [-Werror=implicit-function-declaration]
2023-08-03T12:44:58.6019533Z      Count = InterlockedIncrementUL(&HalpBiosCallCount);
2023-08-03T12:44:58.6020079Z              ^~~~~~~~~~~~~~~~~~~~~~
2023-08-03T12:44:58.6025579Z              InterlockedIncrement
2023-08-03T12:44:58.6026483Z ../src/hal/halx86/generic/x86bios.c:558:5: error: implicit declaration of function 'InterlockedDecrementUL'; did you mean 'InterlockedDecrement16'? [-Werror=implicit-function-declaration]
2023-08-03T12:44:58.6027179Z      InterlockedDecrementUL(&HalpBiosCallCount);
2023-08-03T12:44:58.6027621Z      ^~~~~~~~~~~~~~~~~~~~~~
2023-08-03T12:44:58.6095424Z      InterlockedDecrement16
2023-08-03T12:44:58.6095930Z cc1: all warnings being treated as errors
```

Reply:
Good catch (I just used the same function we have in the kernel, but it's just an internal one). But it does exactly the same kind of thing I had (hides the intermediate cast to PLONG).